File: CS/DXGrid_UnboundColumns_MVVM/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using System.Collections.ObjectModel;

namespace DXGrid_UnboundColumns_MVVM {
    public class ViewModel : ViewModelBase {
        public ObservableCollection<Product> Items { get; set; }
        public ViewModel() {
            Items = ProductList.GetData();
        }

        [Command]
        public void UnboundColumnData(UnboundColumnRowArgs args) {
            if(args.IsGetData) {
                var item = (Product)args.Item;
                args.Value = item.UnitPrice * item.Quantity;
            }
        }
    }
}

File: VB/DXGrid_UnboundColumns_MVVM/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System.Collections.ObjectModel

Namespace DXGrid_UnboundColumns_MVVM

    Public Class ViewModel
        Inherits ViewModelBase

        Public Property Items As ObservableCollection(Of Product)

        Public Sub New()
            Items = ProductList.GetData()
        End Sub

        <Command>
        Public Sub UnboundColumnData(ByVal args As UnboundColumnRowArgs)
            If args.IsGetData Then
                Dim item = CType(args.Item, Product)
                args.Value = item.UnitPrice * item.Quantity
            End If
        End Sub
    End Class
End Namespace
